nft_ingester/src/parsers/mod.rs

```diff
@@ -0,0 +1,73 @@
+mod bubblegum;
+mod crud;
+mod gummy_roll;
+
+pub use self::bubblegum::*;
+pub use self::crud::*;
+pub use self::gummy_roll::*;
+
+use async_trait::async_trait;
+use flatbuffers::{ForwardsUOffset, Vector};
+use plerkle_serialization::account_info_generated::account_info;
+use solana_sdk::pubkey::Pubkey;
+use std::collections::HashMap;
+
+use crate::error::IngesterError;
+use plerkle_serialization::transaction_info_generated::transaction_info::{
+    self, CompiledInstruction,
+};
+
+pub struct ProgramHandlerManager<'a> {
+    registered_parsers: HashMap<Pubkey, Box<dyn ProgramHandler + 'a>>,
+}
+
+impl<'a> ProgramHandlerManager<'a> {
+    pub fn new() -> Self {
+        ProgramHandlerManager {
+            registered_parsers: HashMap::new(),
+        }
+    }
+
+    pub fn register_parser(&mut self, parser: Box<dyn ProgramHandler + 'a>) {
```

Comment:
Overall I really like the ability to register parsers that can be configured to respond to account updates and/or instructions.  I think this design will future proof this code really well.  The `ProgramHandlerManager` struct and `ProgramHandler` trait are pretty cool!